docs/rfcs/0003/readme.md

```diff
@@ -0,0 +1,126 @@
+---
+authors: Matt Richerson <matthew.richerson@hpe.com>
+state: discussion
+---
+Storage Profiles for Lustre Layout
+=============================================
+
+Layout of the Lustre targets can have a large effect on file system performance. Currently, the only parameters exposed for tuning a Lustre file system are for the individual target. This RFC proposes adding fields to the NnfStorageProfile to influence where and how many Lustre targets are allocated.
+
+Proposal
+--------
+
+Users may want Lustre file systems with different performance characteristics. For example, a user job with a single compute node accessing the Lustre file system would see acceptable performance from a single OSS. A FPP workload might want as many OSSs as posible to avoid contention. Currently, the WLM does not have enough information in the DirectiveBreakdown to differentiate these two cases.
+
+The NnfStorageProfile will be modified to allow admins to specify where and how many Lustre targets are allocated by the WLM. An end user then chooses an appropriate storage profile for their workload and includes it on the #DW line. During the proposal phase of the workflow, the Rabbit software will use the information in the NnfStorageProfile to add extra constraints to the allocations in the DirectiveBreakdown. The WLM uses these constraints when picking storage.
+
+The NnfStorageProfile is changed to allow three new fields in the mgtOptions, mdtOptions, and ostOptions. The new fields are:
+
+- count - A static value for how many Lustre targets to use.
+- scale - A value from 1-10 that the WLM can use to determine how many Lustre targets to allocate. This is up to the WLM and the admins to agree on how to interpret this field. I imagine a combination of compute node count and allocation size would be used. Using scale=10 as an example, a single compute node job allocating 10TiB might only get 1 OSS. In contrast, a 5000 compute node job allocating 10 TiB might get 20 OSSs. A 5000 compute node job allocating 100TiB might get 100 OSSs.
+- limit - Limit which storage can be picked. Currently the only values are "None" and "computePhysical". "computePhysical" adds a location constraint in the DirectiveBreakdown that limits the WLM to picking storage with a physical connection to storage. In practice this means that only Rabbit storage in the same chassis as the compute nodes a job is using can be picked. This can be set individually for each of the Lustre target types. When limit is "none", any Rabbit storage can be picked, even if the Rabbit doesn't share a chassis with any of the compute nodes in the job.
```

Comment:
Will this be enforced? I am wondering what the WLM is supposed to do when there are a lot of chassis in the job, a lot more than it makes sense to have targets. If it isn't enforced then I guess this would really just be a question for @behlendorf about what is desirable.

Reply:
We can't enforce this when you fill in the servers resource because we don't have the list of compute nodes yet (or at least we can't trust the data until PreRun). This isn't saying that you have to put one Lustre target on each chassis in the job. It's just limiting you to only pick Rabbits in the chassis that are part of the job. The number of Lustre targets is still determined by the "count" or "scale" values.